source/nodejs/adaptivecards-designer/src/card-designer-surface.ts

```diff
@@ -807,6 +810,10 @@ export class CardDesignerSurface {
 
     startDrag(peer: DesignerPeers.DesignerPeer) {
         if (!this.draggedPeer) {
+			if (peer instanceof DesignerPeers.CarouselPeer) {
+				this.reassignCardElementToCarousel(peer);
```

Comment:
I'm not getting this. If there is no dragged peer and the peer that the user is starting to drag is a `CarouselPeer`, then  move all of the root peer's children that are not `CarouselPeers` into the first page of the Carousel being dragged?

That doesn't seem to make much sense to me, but I might be missing something.

Reply:
For now, carousel is a singleton element. When you add a new carousel, the existing card elements are added to the first carouselPage so we maintain this behavior.

Reply:
I see. Fair enough.